### PR TITLE
Add tests and better null handling for isThrowableCause

### PR DESCRIPTION
**Pull request by stephendavidmarsh**
This PR changes the behavior of `isThrowableCause` to better handle exceptions with `null` as their cause (all exceptions not created with an explicit cause). Instead of producing a null pointer exception, it will produce an appropriate "does not match" message.

Tests  are also added for `isThrowableCause`.

### CODE REVIEW

File: test-tests/shared/src/test/scala/zio/test/AssertionSpec.scala [hunk 1]
```diff
@@ -569,7 +569,21 @@ object AssertionSpec extends ZIOBaseSpec {
       val assertionM = AssertionM.assertionDirect[Unit]("sameName")()(_ => ???)
       assert(assertion.equals(assertionM))(isFalse ?? "assertion != assertionM") &&
       assert(assertionM.equals(assertion))(isFalse ?? "assertionM != assertion")
-    }
+    },
+    test("isThrowableCause must succeed when supplied value has matching cause") {
```
Comment (adamgfraser) [inline test-tests/shared/src/test/scala/zio/test/AssertionSpec.scala:573]: "hasThrowableCause" here?

**Review by adamgfraser — CHANGES_REQUESTED** [review]
One very minor comment but otherwise looks great! Thank you for this!

File: test-tests/shared/src/test/scala/zio/test/AssertionSpec.scala (hunk 1 above)
Comment (stephendavidmarsh) [reply test-tests/shared/src/test/scala/zio/test/AssertionSpec.scala]: :facepalm: Yep, hasThrowableCause, fixed in the commit below.

Comment (adamgfraser) [reply test-tests/shared/src/test/scala/zio/test/AssertionSpec.scala]: Awesome!

**Review by adamgfraser — APPROVED** [review]

**Review by adamgfraser — APPROVED** [review]
